### ISSUE TITLE
Use "modern client" and add path to "router base", and use nuxt generate, the JS can't be loaded.

### ISSUE THREAD

**Issue by ycs77** [nuxt/nuxt.js#5198]
### Version

[v2.4.5](https://github.com/nuxt.js/releases/tag/v2.4.5)

### Reproduction link

[Github](https://github.com/ycs77/nuxt-test-gen)
[GH page](https://ycs77.github.io/nuxt-test-gen/)

### Steps to reproduce

```
git clone git@github.com:ycs77/nuxt-test-gen.git
cd nuxt-test-gen
yarn install
yarn generate
```

> Use server start site in the "dist" directory
> Open browser DevTools > Network
> You can look the JS is not loaded

### What is expected ?

The site JS loaded successfully.

### What is actually happening?

*nuxt.config.js*
```
module.exports = {
  modern: 'client',
  router: {
    base: process.env.DEPLOY_ENV === 'gh_pages' ? '/nuxt-test-gen/' : ''
  },
  ...
}
```
If use one of a "modern client" or "router base", nothing will happen.
But if you use it as above, the site JS will not loaded.

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c8801">#c8801</a>)</em></sub></div>

**Comment by clarkdo** [comment]
@ycs77 Thanks for reporting this bug.

This issue should be fixed in #5202 , but one thing need to be noticed is:

`router.base` needs to be same value between `generate` and `start`, so please add `cross-env DEPLOY_ENV=gh_pages ` in `start` and then access `http://localhost:3000/nuxt-test-gen/`

**Comment by ycs77** [comment]
Thank you for your reminder
